Content.Shared/Projectiles/ProjectileComponent.cs

```diff
@@ -41,5 +42,20 @@ public sealed partial class ProjectileComponent : Component
     [DataField("soundForce")]
     public bool ForceSound = false;
 
+    /// <summary>
+    ///     Whether this projectile will only collide with entities if it was shot from a gun (<see cref="WasFired"/>)
+    /// </summary>
+    [DataField("onlyCollideWhenShot")]
+    public bool OnlyCollideWhenShot = false;
+
+    /// <summary>
+    ///     Whether this projectile has already damaged an entity.
+    /// </summary>
     public bool DamagedEntity;
+
+    /// <summary>
+    ///     If this projectile was fired from a gun or not.
+    /// </summary>
+    [DataField("wasFired")]
+    public bool WasFired;
```

Comment:
Couldnt you deduce this by checking whether Weapon is null? (could rename it by FiredByWeapon or something if thats the case)

Reply:
oh, probably? ill check

Reply:
yea this works